pep-0701.txt

```diff
@@ -0,0 +1,367 @@
+f-strings could be implemented into CPython without modifying the existing
+lexer. These limitations have been recognized previously and previous attempts
+have been made to lift them in :pep:`536` but non of this work was ever implemented
+(see [2]_). Some of these limitations (collected originally by :pep:`536`) are:
+
+#. It is impossible to use the quote character delimiting the f-string
+   within the expression portion::
+
+    >>> f'Magic wand: { bag['wand'] }'
+                                 ^
+    SyntaxError: invalid syntax
+
+#. A previously considered way around it would lead to escape sequences
+   in executed code and is prohibited in f-strings::
+
+    >>> f'Magic wand { bag[\'wand\'] } string'
+    SyntaxError: f-string expression portion cannot include a backslash
+
+#. Comments are forbidden even in multi-line f-strings::
+
+    >>> f'''A complex trick: {
+    ... bag['bag']  # recursive bags!
+    ... }'''
+    SyntaxError: f-string expression part cannot include '#'
+
+#. Arbitrary nesting of expressions without expansion of escape sequences is
+   available in every single other language employing a string interpolation
+   method that uses expressions instead of just variable names. [6]_
+
+These limitations serve no purpose from a language user perspective and
+can be lifted by giving f-literals a regular grammar without exceptions
+and implementing it using dedicated parse code.
+
+The other problems that f-strings have is that the current implementation in
+CPython relies on tokenising f-strings as `STRING` tokens and a post processing of
+these tokens. This has the following problems:
+
+#. It adds a considerable maintainance cost to the CPython parser. This is because
+   the parsing code needs to be written by hand, which has historically lead to a
+   considerable number of inconsistencies and bugs. Writing and maintaining parsing
+   code by hand in C has always been considered error prone and dangerous as it needs
+   to deal with a lot of manual memory management over the original lexer buffers.
+
+#. The f-string parsing code is not able to use the new improved error message mechanisms
+   that the new PEG parser, originally introduced in :pep:`617`, has allowed. The
+   improvements that these error messages brought has been greatly celebrated but
+   unfortunately f-strings cannot benefit from them because they are parsed in a
+   separate piece of the parsing machinery. This is especially unfortunate, since
+   there are several syntactical features of f-strings that can be confusing due
+   to the different implicit tokenization that happens inside the expression
+   part (for instance ``f"{y:=3}"`` is not an assignment expression).
+
+#. Other Python implementations have no way to know if they have implemented
+   f-strings correctly because contrary to other language features, they are not
+   part of the official Grammar [1]_. This is important because several prominent
```

Comment:
```suggestion
   part of the :ref:`official Python grammar <f-strings>`.
   This is important because several prominent
```

Use a direct Intersphinx reference rather than a manual hardcoded URL